R/utils-time.R

```diff
@@ -13,30 +13,21 @@
 #' @importFrom lubridate hours
 #' @export
 flexibleTimeParse <- function(x, timezone, shiftTimeToNoon = TRUE) {
+  format_str <- c("Ymd HMOS z", "Ymd T* z*", "Ymd HMS")
   
-  #first attempt utc
-  format_str <- "Ymd HMOS z"
   time <- parse_date_time(x,format_str, tz=timezone,quiet = TRUE)
   
-  #then attempt an offset time
-  if(isEmptyOrBlank(time)) {
-    format_str <- "Ymd T* z*"
-    time <- parse_date_time(x,format_str, tz=timezone, quiet = TRUE)
-  }
+  dvTimes <- x[which(is.na(time))]
```

Comment:
since you extract DVs here and then put them back in with `time[which(is.na(time))] <- dv`, it would be simpler to just perform the operations on the NAs. Also, I think `isEmptyOrBlank` should still be used, but we want a specific T/F for each element so we use sapply. Then, you can eliminate the if structure. I think the entire function could become this:

```r
# Try these formats on each of the dates. It will try all formats given in 
# format_str. If none work, it will return NA.
format_str <- c("Ymd HMOS z", "Ymd T* z*", "Ymd HMS")
time <- parse_date_time(x,format_str, tz=timezone,quiet = TRUE)

#Handle DVs
#Try the DV format for dates that are still NA
#Add noon as the time if necessary
which.still.NA <- sapply(time, repgen:::isEmptyOrBlank)
format_str_DV <- "Ymd"
time[which.still.NA] <- parse_date_time(x[which.still.NA],format_str_DV, tz=timezone,quiet = TRUE)
if (shiftTimeToNoon) {
    time[which.still.NA] <- time[which.still.NA] + hours(12)
}

return(time)
```